polyfill/lib/zoneddatetime.mjs

```diff
@@ -224,6 +224,9 @@ export class ZonedDateTime {
     let { year, month, day, hour, minute, second, millisecond, microsecond, nanosecond } =
       ES.InterpretTemporalDateTimeFields(calendar, fields, options);
     const offsetNs = ES.ParseOffsetString(fields.offset);
```

Comment:
Why doesn't `ES.ParseOffsetString` thrown, like `ParseTimeZoneOffsetString` in the spec?

Reply:
Not sure, it was before my time. I suspect it's because the polyfill uses this function both for parsing ("get me an offset") and for validation ("is this a valid offset?").  For example, `GetCanonicalTimeZoneIdentifier` calls `ParseOffsetString`, and if it fails then it falls back to calling Intl to get the canonical IANA time zone.

Instead of keeping the return-null-on-failure behavior, another choice would be to throw (like the spec does) and use a try/catch for validation. I don't have an opinion either way, just want to fix the bug.

Reply:
I guess because there wasn't a convenient way to express "If _offsetString_ does not satisfy the syntax of a _TimeZoneNumericUTCOffset_" in code, although I would certainly not mind refactoring the code to make one so that the spec and code agreed better.

Reply:
Let me know how you think we should handle this. Seems like there are three options:
1. Leave the current behavior in this PR where null is checked-for after `ES.ParseOffsetString` returns
2. Change `ES.ParseOffsetString` to throw exceptions, and change usage of `if (ES.ParseOffsetString(s) === null)` to use try/catch instead.
3. Change `ES.ParseOffsetString` to throw exceptions, export a new function from ecmascript.mjs that will return the result of `OFFSET.test()`, and use it in the places where failing to parse should NOT throw an exception, e.g. `TimeZone.p.getNextTransition`.

If I had to choose, (3) seems closest to the spec text and throws fewer exceptions (which is often easier for debugging) so I'd probably choose that one. But any of them are OK with me.

@ptomato feel free to chime in if you have an opinion too.

Reply:
Any of these are OK with me too but I have somewhat of a preference for (3).

Reply:
Me too. I'll plan to implement that unless anyone has another preference.

Reply:
(3) is implemented. New test-only method is called `TestTimeZoneOffsetString`.  Also, after I added throwing to the parsing method, I renamed it to `ParseTimeZoneOffsetString` because that's what it's called in the spec and the polyfill behavior now matches the spec behavior.